.github/workflows/main.yml

```diff
@@ -0,0 +1,104 @@
+name: taisei-build-tests
+on:
+  push:
+    branches:
+      - master
+    tags:
+      - v1.*
+    paths:
+      - "**.c"
+      - "**.h"
+      - "**.yml"
+      - "**.yaml"
+      - "**.build"
+  pull_request:
+    paths:
+      - "**.c"
+      - "**.h"
+      - "**.yml"
+      - "**.yaml"
+      - "**.build"
+
+jobs:
+  linux-test-build:
+    if: "!contains(github.event.head_commit.message, '[skip ci]')"
```

Comment:
What happens if a push contains several commits, and only one of them has `[skip ci]`? Ideally we want to run CI unless *all* of the pushed commits have the skip tag.

Reply:
`*.head_commit.*` only looks at the top-most `HEAD` commit, rather than the entire stack of commits. So any commit after a commit with `[skip ci]` will still run so long as it doesn't contain that string.

Reply:
One thing to note, actually, is that `github.events.*` only apply to non-PR actions, so commits with `[skip ci]` will still run in PRs. A bit annoying but if you're putting up a PR it should probably run every time.